integration_tests/src/main/python/data_gen.py

```diff
@@ -416,11 +416,16 @@ def gen_part_double():
 
 class BooleanGen(DataGen):
     """Generate Bools (True/False)"""
-    def __init__(self, nullable=True):
+    def __init__(self, nullable=True, const_value=None):
         super().__init__(BooleanType(), nullable=nullable)
+        self.const_value = const_value
 
     def start(self, rand):
-        self._start(rand, lambda : bool(rand.getrandbits(1)))
+        if self.const_value:
```

Comment:
This does not generate a constant value if `const_value` is `False`.  It will fall through to the random generator instead.

I don't think we need to modify the data generators for this.  There's already a `SetValuesGen` we should be able to use via something like this in the respective tests:
```python
  bool_gen = boolean_gen if const_bool is None else SetValuesGen(BooleanType(), [const_bool])
```

Reply:
Nice catch. Updated